src/buildstream/buildelement.py

```diff
@@ -228,14 +228,23 @@ def configure_dependencies(self, dependencies):
             # Determine the location to stage each element, default is "/"
             location = "/"
             if dep.config:
-                dep.config.validate_keys(["location"])
+                dep.config.validate_keys(["digest-environment", "location"])
                 location = dep.config.get_str("location")
             try:
                 element_list = self.__layout[location]
             except KeyError:
                 element_list = []
                 self.__layout[location] = element_list
 
+            # Determine the optional digest environment variable to stage each
+            # element,default is None
+            digest_environment = None
+            if dep.config:
+                dep.config.validate_keys(["digest-environment", "location"])
+                if "digest-environment" in dep.config:
+                    digest_environment = dep.config.get_str("digest-environment")
+            self.__digest_environment = digest_environment
```

Comment:
This should  be a dict so as to allow more than one element to set the new config option.